### PR TITLE
refactor(revision): replace Data with concrete types

### PR DESCRIPTION
**Pull request by TWT233**
currently, Revision.Data is an interface{}, which make code unreadable and hard to maintain.

there is an alternative design proposed in #135: individually extract Revision types, and fill Data with concrete type accordingly. In #135, it is only applied on CharacterRevision.

this commit extends the design to all types of Revision

### CODE REVIEW

**Comment by TWT233** [comment]
https://github.com/bangumi/server/pull/164/files#diff-91cf36f4c4d869a61e00df2c963bd9408bd54a2db854a56b0c750d5cf46dac85L333-L338

这个遗留问题算是顺便解决了，不过我没有构造用例来测试（没细看subject revision怎么构造）

File: internal/web/handler/revision.go [hunk 1]
```diff
@@ -302,52 +295,48 @@ func convertModelPersonRevision(r *model.Revision, creatorMap map[model.UIDType]
...
-				VoteField:    subjectData.VoteField,
-				FieldInfobox: subjectData.FieldInfobox,
-				FieldSummary: subjectData.FieldSummary,
-				Platform:     subjectData.Platform,
-				TypeID:       subjectData.TypeID,
-				SubjectID:    subjectData.SubjectID,
-				FieldEps:     subjectData.FieldEps,
-				Type:         subjectData.Type,
-			}
+	if r.Data != (*model.SubjectRevisionData)(nil) {
```
Comment (Trim21) [inline internal/web/handler/revision.go:327]: `r.Data` 不是interface的话应该可以直接跟nil比较吧？

File: internal/model/revision.go [hunk 2]
```diff
@@ -67,28 +74,15 @@ type SubjectRevisionData struct {
...
-	Summary   string
-	ID        uint32
-	CreatorID uint32
-	Type      uint8
-}
-
 // CharacterRevision concrete revision data type.
 type CharacterRevision struct {
 	Data CharacterRevisionData
 	RevisionCommon
```
Comment (Trim21) [inline internal/model/revision.go:80]: `Common: RevisionCommon` 比较好吧？

File: internal/web/handler/revision.go (hunk 1 above)
Comment (TWT233) [reply internal/web/handler/revision.go:327]: 记不得了

不过应该是可以的，因为这里有具体的rtype可以用

File: internal/model/revision.go (hunk 2 above)
Comment (TWT233) [reply internal/model/revision.go:80]: 用 embed 的原因是为了尽量贴合原来的结构使用习惯，比如原来的`revision.Name`到现在还讲得通，不用写`revision.Common.Name`

File: internal/web/handler/revision.go (hunk 1 above)
Comment (Trim21) [inline internal/web/handler/revision.go:327]:
```suggestion
	if r.Data != nil {
```

**Pull request merged**